nomad/acl_endpoint.go

```diff
@@ -1716,7 +1716,15 @@ func (a *ACL) UpsertAuthMethods(
 	}
 
 	// Validate each auth method, canonicalize, and compute hash
+	// merge methods in case we're doing an update
 	for idx, authMethod := range args.AuthMethods {
+		// if there's an existing method with the same name, we treat this as
+		// an update
+		existingMethod, _ := stateSnapshot.GetACLAuthMethodByName(nil, authMethod.Name)
+		if existingMethod != nil {
+			authMethod.Merge(existingMethod)
+		}
```

Comment:
I'd think about moving this if-case into the `Merge` implementation - as it is, it's kind of a leaky abstraction

Reply:
Good point.